src/components/cleaner.ts

```diff
@@ -44,23 +45,42 @@ export class Cleaner {
         if (configuration.get('latex.clean.subfolder.enabled') as boolean) {
             globs = globs.map(globType => './**/' + globType)
         }
+        globs = Array.from(new Set(globs))  // remove duplicate entries
         this.extension.logger.addLogMessage(`Clean glob matched files: ${JSON.stringify({globs, outdir})}`)
-        const files = globs.map(g => glob.sync(g, {cwd: outdir}))
-            // Reduce the array of arrays to a single array containing all the files that should be deleted
-            .reduce((all, curr) => all.concat(curr), [])
-            // Resolve the absolute filepath for every file
-            .map(file => path.resolve(outdir, file))
-        for (const file of files) {
+        // All glob patterns for folders, the glob should be end with a slash (`path.sep`) and the last component should not contain `**`
+        const folderGlobs: string[] = globs.filter(globType => globType.endsWith(path.sep) && !path.basename(globType).includes('**'))
```

Comment:
Variable names in the following are a bit confusing to me. Recommend to add comments and maybe rename for better understanding, e.g., `pathSet` and `paths` below for whose paths, and what `entries` are.